File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-variable-longest/longest-subarray-sum-k/code/java/LongestSubarraySumKSol2.java
class Solution {
    public int longestSubarray(int[] nums, int k) {
        int l=0, sum=0, max=0;
        for (int r=0;r<nums.length;r++) { sum+=nums[r]; while(sum>k) sum-=nums[l++]; max=Math.max(max,r-l+1); }
        return max;
    }
}

File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-variable-longest/longest-subarray-sum-k/code/cpp/longest_subarray_sum_k_sol2.cpp
int longestSubarray(vector<int>& nums, int k) {
    int left = 0, sum = 0, maxLen = 0;
    for (int right = 0; right < nums.size(); right++) {
        sum += nums[right];                    // Expand window
        while (sum > k) sum -= nums[left++];   // Shrink if exceeds
        maxLen = max(maxLen, right - left + 1);
    }
    return maxLen;
}